banyan/banyan/sessions.py

```diff
@@ -1,8 +1,194 @@
+    if executor_code_last_modified > lambda_last_modified:
+        # Update executor code
+        bucket_name, s3_key = _get_executor_lambda_zip()
+        _s3_client.update_function_code(
+            FunctionName=executor_lambda_function_name,
+            S3Bucket=bucket_name,
+            S3Key=s3_key,
+        )
+
+
+def _create_executor_lambda_iam_role():
+    assume_role_policy = {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Principal": {
+                    "Service": "lambda.amazonaws.com"
+                },
+                "Action": "sts:AssumeRole"
+            },
+        ]
+    }
+    basic_lambda_policy_arn = "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+    iam_role_name = "banyan-executor-lambda-role"
+    try:
+        role = _iam_client.create_role(
+            RoleName=iam_role_name,
+            AssumeRolePolicyDocument=json.dumps(assume_role_policy)
+        )
+        _iam_client.get_waiter("role_exists").wait(RoleName=iam_role_name)
+        _iam_client.attach_role_policy(RoleName=iam_role_name, PolicyArn=basic_lambda_policy_arn)
+        time.sleep(5)
+    except ClientError as error:
+        if error.response["Error"]["Code"] == "EntityAlreadyExists":
+            role = _iam_client.get_role(RoleName=iam_role_name)
+        else:
+            raise
+    return role["Role"]["Arn"]
+
+
+def _create_executor_lambda_function(
+    site_packages_dir: str, version: str, environment_hash: str
+):
+    """Creates an executor Lambda if one doesn't exist with the same hash."""
+    executor_lambda_function_name = (
+        f"executor_python{version.replace('.', '-')}_{environment_hash}"
+    )
+    # Note that there may be race conditions here, since creating a lambda might
+    # take some time.
+    # Check if the function exists, and if it does not, create one
+    try:
+        executor_info = _lambda_client.get_function(
+            FunctionName=executor_lambda_function_name
+        )
+        last_modified = int(
+            convert_iso_time(executor_info["Configuration"]["LastModified"])
+        )
+        _update_executor_code(
+            executor_lambda_function_name=executor_lambda_function_name,
+            lambda_last_modified=last_modified,
+        )
+    except _lambda_client.exceptions.ResourceNotFoundException:    
+        # Zip together directory and create function
+        bucket_name, s3_key = _get_executor_lambda_zip(site_packages_dir)
+        _lambda_client.create_function(
+            FunctionName=executor_lambda_function_name,
+            Runtime=f"python{version}",
+            Role=_create_executor_lambda_iam_role(),
+            Handler="executor.lambda_handler",
+            Code={
+                "S3Bucket": bucket_name,
+                "S3Key": s3_key,
+            },
+            Timeout=900,
+            MemorySize=10240,
```

Comment:
Is 10240 the maximum memory size? it should be

Reply:
Yes, 10,240 MB is 10 GB